Microsoft.DotNet.Interactive.CSharp/CSharpKernel.cs

```diff
@@ -309,12 +301,67 @@ bool ShouldRebuild()
                 context);
         }
 
+        // This error is invoked from within the package manager library, which happens in a seperate
+        // process, we scrape stdio for error messages in RestoreAsync method below.
+        private ResolvingErrorReport ReportError = (ErrorReportType errorType, int code, string message) =>
+        {
+            if (errorType == ErrorReportType.Error)
+            {
+                Console.WriteLine("PackageManagement Error {0} {1}", code, message);
```

Comment:
We shouldn't write directly to the console. Instead, we can publish events: `StandardOutputValueProduced` and `StandardErrorValueProduced`.

Reply:
This callback is invoked from within msbuild.  It produces the errors inline with the msbuild console.  We then scrape the entire dotnet build output text and display it.